File: web1_AnaTanaskovic/web1_AnaTanaskovic/www/cgi-bin/admin-panel.py
#!/usr/bin/python

import sqlite3 
from sqlite3 import Error
import http.cookies
import cgi
import os
import datetime

form = cgi.FieldStorage()
username = form.getvalue("username")

def createconnection(dbfile):
	conn = None
	try:
		conn = sqlite3.connect(dbfile)
	except Error as e:
		print(e)
	return conn

def get_plants():
    curs.execute("SELECT * FROM plants")
    plants = curs.fetchall()
    return plants

def is_admin():
    sql = "SELECT role from users where username=? and role='admin'"
    curs.execute(sql, (username,))
    found_admin = curs.fetchone()
    return True if found_admin else False

dbfile = "plantsdatabase.db"
conn = createconnection(dbfile)	
curs = conn.cursor() 

plants = get_plants()
isAdmin = is_admin()

print("Content-Type: text/html")
print()
print("""
<html>
	<head>
    """)
if not username or not isAdmin:
    print("<meta http-equiv='refresh' content='0; url=login.py'>")
print("""
		<title>Main Page</title>
        <link rel="stylesheet" type="text/css" href="/index.css">
        <link rel="stylesheet" type="text/css" href="/admin-panel.css">
        <script src="/admin-panel.js"></script>
	</head>
	<body>
        <div class="container">
            <div class="header">
                <h1>House of plants</h1>
            </div>
            <div class="menu">
                <div>
""")
print("<a href='index.py?username={}'>Home</a>".format(username))
print("""
                    <a href=''>Admin panel</a>

                </div>
                <a href="login.py" class="logout-button">Logout</a>
            </div>

            <div id='content-container' class='flexRow'>
                <div class="table-container">
                    <a onclick='showAddplantForm()' class='btn add-btn'>Add</a>
                    <table id='plantsTable'>
                        <tr>
                            <th>ID</th>
                            <th>price</th>
                            <th>Name</th>
                            <th>Action</th>
                        </tr>
""")
for plant in plants:
    plant_id = plant[0]
    price = plant[1]
    name = plant[2]

    print("""
                        <tr>
                            <td>{0}</td>
                            <td>{1}</td>
                            <td>{2}</td>
                            <td>
                                <a id='edit-plant-btn' onclick="showEditplantForm({0}, '{1}', '{2}')" class='btn'>Edit</a>
                                <a id='remove-plant-btn' onclick='deleteplant({0})' class='btn'>Delete</a>
                            </td>
                        </tr>
    """.format(plant_id, price, name, username))
print("""
                    </table>
                </div>
                <div id='form-container' class='flexColumn'></div>
            </div>
        </div>
	</body>
</html>
""")

File: web1_AnaTanaskovic/web1_AnaTanaskovic/www/admin-panel.js
function showAddplantForm(){
    var addplantForm = document.getElementById('form-container');
    var content = "";
    content+="<div id='form-container' class='flexColumn'>";
    content+="<form class='flexColumn' action='add_plant.py' method='post'>";
    content+="<h3 style='padding-bottom: 20px;'>Add new plant</h3>";
    content+="<label for='price'>price:</label>";
    content+="<input type='text' id='price' name='price'>";
    content+="<br><br>";
    content+="<label for='name'>Name:</label>";
    content+="<input type='text' id='name' name='name'>";
    content+="<br><br><br>";
    content+="<input class='btn' type='button' onclick='addNewplant()' value='Insert'>";
    content+="</form>";
    addplantForm.innerHTML = content;
}

function showEditplantForm(plant_id, price, name){
    var addplantForm = document.getElementById('form-container');
    var content = "";
    content+="<div id='form-container' class='flexColumn'>";
    content+="<form class='flexColumn' action='edit_plant.py' method='post'>";
    content+="<h3 style='padding-bottom: 20px;'>Edit plant</h3>";
    content+="<input type='hidden' id='editId' name='editId' value='" + plant_id + "'>";
    content+="<label for='price'>price:</label>";
    content+="<input type='text' id='editprice' name='editprice' value='" + price + "'>";
    content+="<br><br>";
    content+="<label for='name'>Name:</label>";
    content+="<input type='text' id='editName' name='editName' value='" + name + "'>";
    content+="<br><br><br>";
    content+="<input class='btn' type='button' onclick='editplant()' value='Update'>";
    content+="</form>";
    addplantForm.innerHTML = content;
}

function addNewplant(){
    var xhr = new XMLHttpRequest();
    xhr.onreadystatechange = function() {
        if (xhr.readyState === XMLHttpRequest.DONE && xhr.status === 200) {
            var plants = JSON.parse(xhr.responseText);
            updateAdminplantsTable(plants);
        }
    };
    xhr.open("GET", "add_plant.py?price=" + document.getElementById('price').value + "&name=" + document.getElementById('name').value);
    xhr.send();
}

function editplant(){
    var xhr = new XMLHttpRequest();
    xhr.onreadystatechange = function() {
        if (xhr.readyState === XMLHttpRequest.DONE && xhr.status === 200) {
            var plants = JSON.parse(xhr.responseText);
            updateAdminplantsTable(plants);
        }
    };
    xhr.open("GET", "edit_plant.py?id=" + document.getElementById('editId').value + "&price=" + document.getElementById('editprice').value + "&name=" + document.getElementById('editName').value);
    xhr.send();
}

function deleteplant(plant_id){
    var xhr = new XMLHttpRequest();
    xhr.onreadystatechange = function() {
        if (xhr.readyState === XMLHttpRequest.DONE && xhr.status === 200) {
            var plants = JSON.parse(xhr.responseText);
            updateAdminplantsTable(plants);
        }
    };
    xhr.open("GET", "remove_plant.py?plant_id=" + plant_id);
    xhr.send();
}

function updateAdminplantsTable(plants){
    var table = document.getElementById('plantsTable');
    var content = "<tr><th>ID</th><th>price</th><th>Name</th><th>Action</th></tr>";
    plants.forEach(plant => {
        content+= "<tr>";
        content+="<td>"+plant[0]+"</td><td>"+plant[1]+"</td><td>"+plant[2]+"</td>"
        content+= "<td><a id='edit-plant-btn' onclick=\"showEditplantForm(" + plant[0] + ", '" + plant[1] + "', '" + plant[2] + "')\" class='btn'>Edit</a> <a id='remove-plant-btn' onclick='deleteplant("+plant[0]+")' class='btn'>Delete</a> </td>";
        content+= "</tr>";
    });
    table.innerHTML = content;
    document.getElementById('form-container').innerHTML = "";
}